src/main/java/org/apache/mesos/hdfs/executor/NodeExecutor.java

```diff
@@ -9,6 +9,13 @@
 import org.apache.mesos.MesosExecutorDriver;
```

Comment:
Same comment as for NameNodeExecutor. This code is too complex and you should only need to add a few lines of health checks scheduled at the appropriate intervals. The existing code should work without too many changes since all that's required for Mesos DNS is a waiting period for DNS registration.